### PR TITLE
Update: Add cwd option for cli-engine (fixes #4472)

### LINKED ISSUE #4472: CLIEngine `cwd` option
It would be useful to be able to set the `cwd` (current working directory) of where ESLint works from. My specific use-case is that I use ESLint in an Atom Editor plugin where `process.cwd() === '/'`.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/28365464-cliengine-cwd-option?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ilyavolodin**
Do we need to add unit tests for this? It's already tested pretty heavily in config.js file. If we do, what should they be?

### CODE REVIEW

**Comment by nzakas** [comment]
We should at least have one test for each that uses a different cwd.

File: docs/developer-guide/nodejs-api.md [hunk 1]
```diff
@@ -134,6 +134,7 @@ The `CLIEngine` is a constructor, and you can create a new instance by passing i
 * `cache` - Operate only on changed files (default: `false`). Corresponds to `--cache`.
 * `cacheFile` - Name of the file where the cache will be stored (default: `.eslintcache`). Corresponds to `--cache-file`. Deprecated: use `cacheLocation` instead.
 * `cacheLocation` - Name of the file or directory where the cache will be stored (default: `.eslintcache`). Correspond to `--cache-location`
+* `cwd` - Path to a directory that should be considered as an current working directory
```
Comment (sindresorhus) [inline docs/developer-guide/nodejs-api.md]: > Path to a directory that should be considered as **the** current working directory**.**

**Comment by ilyavolodin** [comment]
Added a few tests (cache and ignored paths), everything else is already covered. Let me know if you want some additional tests.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**